### ISSUE TITLE
MultiDict lacks a way to filter individual items

### ISSUE THREAD

**Issue by MartinSStewart** [Janiczek/elm-bidict#1]
I have a MultiDict that keeps track of login attempts. It looks like this `MultiDict SessionId Time.Posix`. 

Every once in a while I'd like to clear old login attempts. I thought I could do that using `MultiDict.filter`, but this expects a function that determines if all the login attempts for a `SessionId` should be filtered. I then thought maybe `MultiDict.map` could work since I could change the Set of login attempts by removing the old ones. But `MultiDict.map` operates on individual values so that doesn't work. What I need is some `MultiDict.filterValues : (key -> value -> Bool) -> MultiDict key value -> MultiDict key value` or `MultiDict.mapSet : (key -> Set a -> Set b) -> MultiDict key a -> MultiDict key b`

**Comment by Janiczek** [comment]
Hello @MartinSStewart!

I'm thinking the correct solution to this is indeed a filter function that takes `key -> value -> Bool` instead of `key -> Set value -> Bool`. The original filter is not general enough.

I'm currently inclined to make the general version the default, and either drop the Set one or keep it around as `filterKeys` or something.

```diff
-MultiDict.filter : (key -> Set value -> Bool) -> MultiDict key value -> MultiDict key value
+MultiDict.filter : (key -> value -> Bool) -> MultiDict key value -> MultiDict key value
```

**Comment by MartinSStewart** [comment]
That makes sense to change the default. I don't think the filterKeys function is going to get used often.

**Comment by Janiczek** [comment]
Here you are in 3.0.0, @MartinSStewart : https://package.elm-lang.org/packages/Janiczek/elm-bidict/latest/MultiDict-Assoc#filter

closed in b806836fad4ade7c0b62dfc2c51297a6f58000e4

**Comment by MartinSStewart** [comment]
Awesome, thanks for the quick package release!
